src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/attn_quant.cpp

```diff
@@ -0,0 +1,248 @@
+        v_i32 = _mm512_max_epi32(v_i32, v_zero);
+        _mm512_mask_cvtusepi32_storeu_epi8(a + i, 0xffff, v_i32);
+    }
+#elif defined(HAVE_AVX2)
+    auto v_scale = _mm256_set1_ps(1 / scale);
+    auto v_zp = _mm256_set1_ps(zp);
+    for (; i + vec_len_f32_avx2 <= n; i += vec_len_f32_avx2) {
+        auto v = mm256_uni_loadu_ps(b + i);
+        v = _mm256_fmadd_ps(v, v_scale, v_zp);
+        v = _mm256_round_ps(v, _MM_ROUND_NEAREST);
+        auto v_i32 = _mm256_cvtps_epi32(v);
+
+        auto high4 = _mm256_extractf128_si256(v_i32, 1);
+        auto low4 = _mm256_castsi256_si128(v_i32);
+        auto packed = _mm_packs_epi32(low4, high4);
+        packed = _mm_packus_epi16(packed, packed);
+        _mm_storel_epi64(reinterpret_cast<__m128i*>(a + i), packed);
+    }
+#endif
+    for (; i < n; i++) {
+        float tmp = b[i];
+        a[i] = static_cast<uint8_t>(std::round(tmp / scale + zp));
+    }
+}
+
+template <typename T, typename T2>
+static void attn_quant_mt(const ov::intel_cpu::PlainTensor& k_input,
+                              const ov::intel_cpu::PlainTensor& v_input,
+                              const ov::intel_cpu::PlainTensor& past_k_output,
+                              const ov::intel_cpu::PlainTensor& past_v_output,
+                              const ov::intel_cpu::PlainTensor& past_k_scale_zp,
+                              const ov::intel_cpu::PlainTensor& past_v_scale_zp) {
+    size_t B = k_input.m_dims[0], H = k_input.m_dims[1], L1 = k_input.m_dims[2], S = k_input.m_dims[3];
+    parallel_for3d(B, H, L1, [&](size_t b, size_t h, size_t m) {
+        auto p_k = past_k_scale_zp.ptr<float>(b, h, m);
+        auto p_v = past_v_scale_zp.ptr<float>(b, h, m);
+        quant_u8(past_k_output.ptr<T2>(b, h, m),
+                 k_input.ptr<T>(b, h, m),
+                 S,
+                 p_k[0],
+                 p_k[1]);
+        quant_u8(past_v_output.ptr<T2>(b, h, m),
+                 v_input.ptr<T>(b, h, m),
+                 S,
+                 p_v[0],
+                 p_v[1]);
+    });
+}
+
+void attn_quantkv(const ov::intel_cpu::PlainTensor& k_input,
+                  const ov::intel_cpu::PlainTensor& v_input,
+                  const ov::intel_cpu::PlainTensor& past_k_output,
+                  const ov::intel_cpu::PlainTensor& past_v_output,
+                  const ov::intel_cpu::PlainTensor& past_k_scale_zp,
+                  const ov::intel_cpu::PlainTensor& past_v_scale_zp) {
+    if (k_input.get_precision() == ov::element::f32 && past_k_output.get_precision() == ov::element::u8) {
+        attn_quant_mt<float, uint8_t>(k_input, v_input, past_k_output, past_v_output, past_k_scale_zp, past_v_scale_zp);
+    } else if (k_input.get_precision() == ov::element::bf16 && past_k_output.get_precision() == ov::element::u8) {
+        attn_quant_mt<ov::bfloat16, uint8_t>(k_input, v_input, past_k_output, past_v_output, past_k_scale_zp, past_v_scale_zp);
```

Comment:
Mb I missed smt, but how quant kernel is supposed to handle bf16? Seems like it uses pure fp32 intrinsics

Reply:
When running on SPR, the k,v will be bf16, the function `mmxxx_uni_loadu_ps` will be used to handle different type, it will be like:
https://github.com/openvinotoolkit/openvino/blob/abf4bfe0cf43793c60280d785a85d03de4b6442a/src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp#L33-L40
``